web/src/pages/project/ProjectGrid.tsx

```diff
@@ -11,18 +17,117 @@ import { ProjectSummaryResponse } from '../../sm-api/api'
 interface ProjectGridProps {
     summary: ProjectSummaryResponse
     projectName: string
+    filterValues: Record<string, string>
+    updateFilters: (e: { [k: string]: FormDataEntryValue }) => void
 }
 
-const ProjectGrid: React.FunctionComponent<ProjectGridProps> = ({ summary, projectName }) => {
+const ProjectGrid: React.FunctionComponent<ProjectGridProps> = ({
+    summary,
+    projectName,
+    filterValues,
+    updateFilters,
+}) => {
     const headers = [
         'Family ID',
         ...summary.participant_keys.map((field) => field[1]),
         ...summary.sample_keys.map((field) => field[1]),
         ...summary.sequence_keys.map((field) => `sequence.${field[1]}`),
     ]
 
+    const [tempFilterValues, setTempFilterValues] =
+        React.useState<Record<string, string>>(filterValues)
+
+    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name } = e.target
+        const { value } = e.target
+        setTempFilterValues({
+            ...Object.keys(tempFilterValues)
+                .filter((key) => name !== key)
+                .reduce((res, key) => Object.assign(res, { [key]: tempFilterValues[key] }), {}),
+            ...(value && { [name]: value }),
+        })
+    }
+
+    const onClear = (column: string) => {
+        updateFilters({
+            ...Object.keys(tempFilterValues)
+                .filter((key) => column !== key)
+                .reduce((res, key) => Object.assign(res, { [key]: tempFilterValues[key] }), {}),
+        })
+    }
+
+    const onSubmit = () => {
+        updateFilters(tempFilterValues)
+    }
+
     return (
         <Table celled>
+            <SUITable.Header>
+                <SUITable.Row>
+                    {headers.map((k, i) => {
+                        if (k === 'Sample ID') {
+                            return (
+                                <SUITable.HeaderCell
+                                    key={`filter-${k}-${i}`}
+                                    style={{ borderBottom: 'none' }}
+                                ></SUITable.HeaderCell>
+                            )
+                        }
+                        return (
+                            <SUITable.HeaderCell
+                                key={`filter-${k}-${i}`}
+                                style={{ borderBottom: 'none' }}
+                            >
+                                <div style={{ position: 'relative' }}>
+                                    <div style={{ position: 'absolute', top: 0, right: 0 }}>
+                                        <Popup
```

Comment:
On further review, I don't think I'm happy with this approach - can you bind more context in here (meta or not) so you don't look up k in filter values, I think it's worth being more concrete about how columns relate to filters rather than looking it up based on the name.